src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -20,15 +37,92 @@ public class FindCommandParser implements ParserComplex<FindCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(PersonType personType, String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        if (args.isBlank()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
+        if (personType.equals(PersonType.PATIENT)) {
+            return parsePatient(args);
+        } else if (personType.equals(PersonType.SPECIALIST)) {
+            return parseSpecialist(args);
+        } else {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+    }
+
+    private FindCommand parsePatient(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
+                        PREFIX_TAG, PREFIX_MEDICALHISTORY);
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL,
+                PREFIX_ADDRESS, PREFIX_MEDICALHISTORY);
+
+        List<Predicate<Person>> predicateList = setupPersonPredicates(argMultimap);
```

Comment:
Really like the abstraction of `setupPersonPredicates` here, makes the function a lot easier to understand.